### PR TITLE
DOC: Doc clean-up and additions

### PR DESCRIPTION
**Pull request by thequackdaddy**
It feels like I'm on a doc binge. But writing docs is often a good way to better understand the tool you are using. 

After playing with the website for a bit, I thought to myself that it would be nice if we could clean-up some of the docs. There's some neat features of stats models that I think the docs could advertise more. I cleaned up the docs and removed some complicated instructions for things that are out of date (I doubt any 32 bit windows user is developing, they'll just install from conda or pip wheels) so I removed some things that are old. 

I'm happy to take a stab at adding some other docs--even if just a stub for now--if it can encourage us to organize some of this better. Things like robust covariance are neat and the fact that stats models make it so easy should be a relatively big selling point. Alas, we don't have docs/examples that really show it off, so at least starting it might be a good idea. If there are others, let me know.

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -174,4 +111,4 @@ Optional Dependencies
   docs locally or to use the notebooks.
 * `joblib <http://pythonhosted.org/joblib/>`__ >= 0.9 can be used to accelerate distributed
   estimation for certain models.
-* `jupyter <http://jupyter.org/>`__ is needed to run the notebooks.
\ No newline at end of file
+* `jupyter <http://jupyter.org/>`__ is needed to run the notebooks.
```
Comment (jbrockmendel) [inline docs/source/install.rst]: Wasn't there a thing recently about all-https?  Not sure if that's relevant here.

**Comment by jbrockmendel** [comment]
Most of the rst stuff I'm not qualified to comment on, but +1 on the idea of making a concerted push to improve docs/docstrings.

File: docs/source/install.rst (hunk 1 above)
Comment (thequackdaddy) [reply docs/source/install.rst]: I think that only applies to images/resources auto-loaded by the webpage. Not links for the user.

**Comment by jbrockmendel** [comment]
If you’ve got anything resembling a master TODO list for docs, can you add the GH wiki page to it pls

File: docs/source/faq.rst [hunk 2]
```diff
@@ -5,37 +5,71 @@
...
 
 .. _build-faq:
 
 Why won't statsmodels build?
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+----------------------------
+
+Remember that to build, you must have:
+
+- The appropriate dependencies (numpy, pandas, Cython) installed
```
Comment (josef-pkt) [inline docs/source/faq.rst]: add scipy here

File: docs/source/gee.rst [hunk 3]
```diff
@@ -121,6 +121,7 @@ The distribution families are the same as for GLM, currently implemented are
    InverseGaussian
    NegativeBinomial
    Poisson
+   Tweedie
```
Comment (josef-pkt) [inline docs/source/gee.rst]: Is tweedie allowed in gee?
I guess there is nothing special enough that it shouldn't work

File: docs/source/install.rst [hunk 4]
```diff
@@ -61,87 +69,16 @@ Windows
...
-    set DISTUTILS_USE_SDK=1
-
-Replace `/x64` with `/x86` and `v7.0` with `v7.1` as needed.
+Getting the right compiler is especially confusing for Windows users. Over time,
+Python has been built using a variety of different Windows C compilers.
+`This guide <https://wiki.python.org/moin/WindowsCompilers>`_ should help
+clarify which version of Python uses which compiler by default.
 
+Alternatively, you can use the
+`MinGW-w64 toolchoin <http://mingw-w64.org/doku.php>`__.
```
Comment (josef-pkt) [inline docs/source/install.rst]: This is not supported anymore for python > 3.4

File: docs/source/large_data.rst [hunk 5]
```diff
@@ -0,0 +1,62 @@
+.. module:: statsmodels.base.distributed_estimation
+.. currentmodule:: statsmodels.base.distributed_estimation
+
+Working with "large"-ish data
+=============================
+
+    Big data is like teenage sex: everyone talks about it, nobody really knows
```
Comment (josef-pkt) [inline docs/source/large_data.rst]: I think we don't want to have this analogy in our documentation

File: docs/source/optimization.rst [hunk 6]
```diff
@@ -0,0 +1,140 @@
+.. module:: statsmodels.base.optimizer
+.. currentmodule:: statsmodels.base.optimizer
+
+Optimization
+============
+
+Many of the models we use such as :ref:`GLM <glm>` and
+:ref:`discrete models <discretemod>` allow for the optional selection of a
+scipy optimizer. This can either be defaut or an option based on the model
```
Comment (josef-pkt) [inline docs/source/optimization.rst]: l missing in default

File: docs/source/optimization.rst [hunk 7]
```diff
@@ -0,0 +1,140 @@
+.. module:: statsmodels.base.optimizer
+.. currentmodule:: statsmodels.base.optimizer
+
+Optimization
+============
+
```
Comment (josef-pkt) [inline docs/source/optimization.rst]: I think a preliminary sentence about non-scipy optimization would be helpful
e.g. something like

"statsmodels uses three types of algorithms for the estimation of the parameters of a model. Linear models such as OLS are directly estimated using appropriate linear algebra functions. Some models use specific algorithm such as iteratively reweighted least squares in GLM and RLM, or the most common method for nonlinear optimization is by using the optimizers in scipy/"

File: docs/source/optimization.rst [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+:ref:`discrete models <discretemod>` allow for the optional selection of a
+scipy optimizer. This can either be defaut or an option based on the model
+selected.
+
+statsmodels supports the following optimizers along with keyword arguments
+associated with that specific optimizer:
+
+- ``newton`` - Newton-Raphson iteration. While not directly from scipy, we
+  consider it an optimizer because only the objective function and the score
+  are required parameters.
```
Comment (josef-pkt) [inline docs/source/optimization.rst]: newton also need the hessian

File: docs/source/optimization.rst [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+      minimizer : dict
+          Extra keyword arguments to be passed to the minimizer
+          `scipy.optimize.minimize()`, for example 'method' - the
+          minimization method (e.g. 'L-BFGS-B'), or 'tol' - the
+          tolerance for termination. Other arguments are mapped from
+          explicit argument of `fit`:
+          - `args` <- `fargs`
+          - `jac` <- `score`
+          - `hess` <- `hess`
+
```
Comment (josef-pkt) [inline docs/source/optimization.rst]: this is missing our newer generic interface to scipy `minimize` which allows to use any of the scipy optimization algorithms.

File: docs/source/faq.rst (hunk 2 above)
Comment (thequackdaddy) [reply docs/source/faq.rst]: Done.

File: docs/source/gee.rst (hunk 3 above)
Comment (thequackdaddy) [reply docs/source/gee.rst]: I would think so. Granted, I'm no expert here.

File: docs/source/install.rst (hunk 4 above)
Comment (thequackdaddy) [reply docs/source/install.rst]: I'll just remove this. There's tons of support now for the "official" C compiler for Windows nowadays, we should aim people towards the most common and supported tools.

File: docs/source/large_data.rst (hunk 5 above)
Comment (thequackdaddy) [reply docs/source/large_data.rst]: I love that quote... but fair enough. I sanitized it.

File: docs/source/optimization.rst (hunk 7 above)
Comment (thequackdaddy) [reply docs/source/optimization.rst]: Sure. You'll have to help me if you want to go through all of them. I don't know much (anything) on time series and all the modules there and I think that's pretty scipy-heavy, too.

File: docs/source/optimization.rst (hunk 8 above)
Comment (thequackdaddy) [reply docs/source/optimization.rst]: Typo. Thanks. I meant to say it just needs score and hessian... It doesn't require objective function (loglike) which is super-helpful it turns out for Tweedie where loglike is difficult.

File: docs/source/optimization.rst (hunk 9 above)
Comment (thequackdaddy) [reply docs/source/optimization.rst]: I saw that, but I honestly have never experimented with it or even tested it. Is there a simple example for this? It sounds useful.

File: docs/source/optimization.rst (hunk 7 above)
Comment (josef-pkt) [reply docs/source/optimization.rst]: I just want us to mention the type of optimization methods without providing a full list of models.

(actually it's not that difficult, skipping some models in the sandbox
linear regression OLS/WLS/GLS and VAR and VECM have closed for linalg estimators, 
only RLM and GLM have IRLS, 
all others have only scipy optimizers + our newton.
GLM is currently the only one with irls and scipy optimizers.

There are some hybrid versions like GLSAR that we don't mention.

... maybe there would be more in a full list, e.g. fit_regularized in OLS/WLS and in GLM uses coordinate descend. 
)

**Comment by thequackdaddy** [comment]
(I seem to have overwritten this accidentally. I need to get on my other computer re-push the correct branch.)

**Comment by thequackdaddy** [comment]
Ok. Fixed the overwritten branch by re-pushing the correct branch.

**Pull request closed without merging**